src/types.rs

```diff
@@ -1,6 +1,47 @@
+use std::fmt;
+
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
+#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, JsonSchema)]
+pub struct HumanAddr(pub String);
+
+#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, JsonSchema)]
+pub struct CanonicalAddr(pub Vec<u8>);
+
+impl HumanAddr {
+    pub fn as_bytes(&self) -> &[u8] {
```

Comment:
I think this API is misleading as it unclear what binary representation comes out of that. I'd prefer to write `.as_str().as_bytes()`.